### PR TITLE
Feature/find times

### LINKED ISSUE #41: As a tandem engineer, I want to take the availability of all parties and find the best time to meet so that the front-end can visualize it.

### PR DESCRIPTION
**Pull request by thejinxters**
closes #41

### CODE REVIEW

File: server/controllers/meeting_controller.coffee [hunk 1]
```diff
@@ -45,8 +49,11 @@ meetingController =
           index = emails.indexOf email
           emails.splice(index, 1)
       Meeting.methods.update(meeting_id, {emails:emails})
-      response.schedule = dummy_response
-      res.status(200).send response
+      buildMeetingCalendar emails, (users, availability)->
```
Comment (stephenthoma) [inline server/controllers/meeting_controller.coffee]: `buildMeetingCalendar emails, (users, availability) ->`

Comment (thejinxters) [inline server/controllers/meeting_controller.coffee]: nice catch

Comment (thejinxters) [inline server/controllers/meeting_controller.coffee]: @sthoma We should make another story about sending back the meeting initiator's schedule with this data

**Comment by stephenthoma** [comment]
So right now, it looks like if there are no tandem users invited to the meeting we're sending an availability calendar of all available. This should probably be changed to completely unavailable

**Comment by thejinxters** [comment]
I think it should be my schedule availability as a meeting initiator instead of none available.

**Comment by stephenthoma** [comment]
Agreed :+1:

File: server/controllers/meeting_controller.coffee [hunk 2]
```diff
@@ -72,58 +79,130 @@ meetingController =
...
+  newTime =  moment(now).add(fifteenMinutes)
+  fifteenMinRange = moment.range(now, newTime)
+
+  #retrieve relevant calendar chunks
+  calendarChunks = createWeekCalendarChunks()
+
+  availableRanges = []
+  for day in calendarChunks
+    dayObj =
+      day_code:day.day_code
```
Comment (stephenthoma) [inline server/controllers/meeting_controller.coffee]: `day_code: day.day_code`

File: server/controllers/meeting_controller.coffee [hunk 3]
```diff
@@ -72,58 +79,130 @@ meetingController =
...
+        timeRange.by fifteenMinRange, (time) ->
+          newRange = moment.range(time, moment(time).add(duration, 'minutes'))
+          if isTimeRangeAvailble(newRange, busyRanges)
+            dayObj[key].push(newRange)
+
+    availableRanges.push(dayObj)
+
+  return availableRanges
+
+createWeekCalendarChunks = ()->
```
Comment (stephenthoma) [inline server/controllers/meeting_controller.coffee]: `createWeekCalendarChunks = () ->`

**Comment by thejinxters** [comment]
:+1:  Looks, good, lets merge :rocket:

**Pull request merged**